t/lib/TestDB.pm

```diff
@@ -12,11 +12,37 @@ sub new {
 
     open my $fh, '<', 'share/test_data.sql';
 
+    my $continued_line = "";
+    my $last_line_was_continued = 0;
+
     while (<$fh>) {
         $ENV{DBIC_TRACE}
           and warn $_;
-
-        $schema->storage->dbh_do( sub { $_[1]->do($_) } );
+	my $line = $_;
+
+	# Remove comments from SQL-File
+	my $comment_start = index ($line, '--');
+	$line = substr ($line, 0, $comment_start);
```

Comment:
Isn't this silently removing the last character on every line that has no comment? `index` returns -1 in this case.

Use a replacement regexp to do this a bit more Perl style:

    # remove comments from SQL file
    $line =~ s/ -- .* $ //x;

Read: search for `--` followed by any character `.` zero or more times `*` and end of line `$` and replace with the empty string. The `/x` modifier is for allowing insignificant whitespace between regexp elements.